// @octokit/action/dist-node/index.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var core = require('@octokit/core');
var authAction = require('@octokit/auth-action');
var pluginPaginateRest = require('@octokit/plugin-paginate-rest');
var pluginRestEndpointMethods = require('@octokit/plugin-rest-endpoint-methods');
var httpsProxyAgent = require('https-proxy-agent');

const VERSION = "5.0.5";

const DEFAULTS = {
  authStrategy: authAction.createActionAuth,
  baseUrl: getApiBaseUrl(),
  userAgent: `octokit-action.js/${VERSION}`
};
function getProxyAgent() {
  const httpProxy = process.env["HTTP_PROXY"] || process.env["http_proxy"];
  if (httpProxy) {
    return new httpsProxyAgent.HttpsProxyAgent(httpProxy);
  }
  const httpsProxy = process.env["HTTPS_PROXY"] || process.env["https_proxy"];
  if (httpsProxy) {
    return new httpsProxyAgent.HttpsProxyAgent(httpsProxy);
  }
  return undefined;
}
const Octokit = core.Octokit.plugin(pluginPaginateRest.paginateRest, pluginRestEndpointMethods.legacyRestEndpointMethods).defaults(function buildDefaults(options) {
  return {
    ...DEFAULTS,
    ...options,
    request: {
      agent: getProxyAgent(),
      ...options.request
    }
  };
});
function getApiBaseUrl() {
  /* istanbul ignore next */
  return process.env["GITHUB_API_URL"] || "https://api.github.com";
}

exports.Octokit = Octokit;
